src/smallmap_gui.cpp

```diff
@@ -1546,10 +1542,9 @@ class SmallMapWindow : public Window {
 							return;
 						}
 						/* Non-fixed legend entries for the owner view. */
-						SetDParam(0, tbl->company);
-						str = STR_SMALLMAP_COMPANY;
+						str = GetString(STR_SMALLMAP_COMPANY, tbl->legend);
```

Comment:
Did you mean to change this?

```suggestion
						str = GetString(STR_SMALLMAP_COMPANY, tbl->company);
```

Reply:
I'd like to rework this legend code at some point, but I think that's a side-quest for another day.